File: Solutions/1053. Previous Permutation With One Swap/1053. Previous Permutation With One Swap.py
class Solution:
    def prevPermOpt1(self, arr: List[int]) -> List[int]:
        idx,n=-1,len(arr)
        if n<=1:
            return arr
        for i in range(n-1,0,-1):
            if arr[i-1]>arr[i]:
                idx=i-1
                break
        if idx==-1:
            return arr
        for i in range(n-1,idx,-1):
            if arr[idx]>arr[i] and arr[i]!=arr[i-1]:
                arr[idx],arr[i]=arr[i],arr[idx]
                break
        return arr

File: Solutions/1053. Previous Permutation With One Swap/1053. Previous Permutation With One Swap.go
func prevPermOpt1(arr []int) []int {
    n,idx:=len(arr),-1
    if n<=1{
        return arr
    }
    for i:=n-1;i>0;i-=1{
        if arr[i]<arr[i-1]{
            idx=i-1
            break
        }
    }
    if idx==-1{
        return arr
    }
    for i:=n-1;i>idx;i-=1{
        if arr[idx]>arr[i] && arr[i]!=arr[i-1]{
            arr[idx],arr[i]=arr[i],arr[idx]
            break
        }
    }
    return arr
}